sphinx/ext/autosummary/generate.py

```diff
@@ -499,10 +500,13 @@ def generate_autosummary_docs(
     suffix: str = '.rst',
     base_path: str | os.PathLike[str] | None = None,
     imported_members: bool = False,
-    app: Any = None,
+    app: Sphinx | None = None,
     overwrite: bool = True,
     encoding: str = 'utf-8',
-) -> None:
+) -> list[Path]:
+    """Generate autosummary documentation for the given sources."""
```

Comment:
You should also indicate what is being returned. 

Now that I think about, I wonder whether returning strings is preferrable or not since you still downcast them to strings anyway (the same could be said for your previous PR).

Reply:
> since you still downcast them to strings anyway

No they don't get downcast to strings

Reply:
> [str(f) for f in new_files] (L599)

Reply:
(Ok, the comment was not at the correct place)

Reply:
> (Ok, the comment was not at the correct place)

right so you mean`new_files` not `all_files` (which is what gets returned)

Reply:
anyway, documented the return value